src/main/scanLogic/scanRunners/binaryRunner.ts

```diff
@@ -42,48 +43,34 @@ export class NotEntitledError extends Error {
  */
 export abstract class BinaryRunner {
     protected _runDirectory: string;
-    private _isSupported: boolean = true;
 
-    protected static readonly RUNNER_FOLDER: string = 'analyzer-manager';
     private static readonly RUNNER_NAME: string = 'analyzerManager';
+    private static readonly RUNNER_FOLDER: string = path.join(ScanUtils.getIssuesPath(), BinaryRunner.RUNNER_NAME);
 
     private static readonly NOT_ENTITLED: number = 31;
 
+    private static readonly DOWNLOAD_URL: string = '/xsc-gen-exe-analyzer-manager-local/v1/[RELEASE]/';
+
     constructor(
         protected _connectionManager: ConnectionManager,
         protected _abortCheckInterval: number,
         protected _logManager: LogManager,
-        protected _binaryPath: string = path.join(ScanUtils.getHomePath(), BinaryRunner.RUNNER_FOLDER, BinaryRunner.getBinaryName())
+        protected _binary: Resource = BinaryRunner.getAnalyzerManagerResource(_logManager)
     ) {
-        this._runDirectory = path.dirname(_binaryPath);
-        this._isSupported = this.validateSupported();
+        this._runDirectory = path.dirname(this._binary.fullPath);
 
         if (this._abortCheckInterval <= 0) {
             // Default check in 1 sec intervals
             this._abortCheckInterval = 1 * 1000;
         }
     }
 
-    /**
-     * Get the binary name for the runner base on the running os
-     * @returns the name of the expected binary file to run
-     */
-    protected static getBinaryName(): string {
-        let name: string = BinaryRunner.RUNNER_NAME;
-        switch (os.platform()) {
-            case 'win32':
-                return name + '_windows.exe';
-            case 'linux':
-                return name + '_linux';
-            case 'darwin':
-                name += '_mac';
-                if (os.arch() === 'arm' || os.arch() === 'arm64') {
-                    return name + '_arm';
-                } else {
-                    return name + '_amd';
-                }
-        }
-        return name;
+    public static getAnalyzerManagerResource(logManager: LogManager): Resource {
+        return new Resource(
+            Utils.addZipSuffix(BinaryRunner.DOWNLOAD_URL + Utils.getArchitecture() + '/' + BinaryRunner.RUNNER_NAME),
```

Comment:
Shouldn't we add `.exe` for Windows in the source too?

Reply:
no, we are downloading zip file, the actual binary exists inside the zip